li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaFederatedConsumerImpl.java

```diff
@@ -260,27 +296,38 @@ synchronized public void assign(Collection<TopicPartition> partitions) {
       throw new IllegalArgumentException("Topic partition collection to assign to cannot be null");
     }
 
-    // if partitions are empty, it should be treated the same as unsubscribe(). This
-    // is the vanilla Kafka consumer behavior.
+    // If partitions are empty, it should be treated the same as unsubscribe(). This is the vanilla Kafka consumer
+    // behavior.
     if (partitions.isEmpty()) {
       unsubscribe();
       return;
     }
 
-    Map<ClusterDescriptor, Collection<TopicPartition>> clusterToTopicPartitionsMap = getPartitionsByCluster(partitions);
+    // TODO: Throw an exception if there exists a partition whose topic exists but the specified partition number is
+    // out of range for that topic. Whether of not to throw an exception can be controlled by a property.
+    Set<TopicPartition> partitionsSet = new HashSet<>(partitions);
```

Comment:
we need this copy because the method signature now expects a Set. this is a potentially significant mem copy - why not accept a collection ?

Reply:
I spent a whole day debugging mock issues involving a mix of Collection and Set in my test code and decided to use a set for assignment set internally. I assumed assign() should not be a frequent operation and a copy should be ok?